apps/client/src/components/ButtonsHeader/ButtonsHeader.tsx

```diff
@@ -15,8 +15,16 @@ interface ButtonsHeaderProps {
 
 export default function ButtonsHeader({ items }: ButtonsHeaderProps) {
   const navigate = useNavigate();
+  const location = useLocation(); 
   const [tab, setTab] = useState(0);
 
+  useEffect(() => {
+    const currentTab = items.findIndex(item => location.pathname.startsWith(item.url));
+    if (currentTab !== -1) {
+      setTab(currentTab);
+    }
+  }, [location.pathname, items]);
+
```

Comment:
This causes a useless render. it would be more efficient to compute the `tab` from a `useMemo`, then changing tab by altering the url.